src/shared/watchdog.c

```diff
@@ -20,6 +21,7 @@ static char *watchdog_device;
 static usec_t watchdog_timeout; /* 0 → close device and USEC_INFINITY → don't change timeout */
 static usec_t watchdog_pretimeout; /* 0 → disable pretimeout and USEC_INFINITY → don't change pretimeout */
 static usec_t watchdog_last_ping = USEC_INFINITY;
+static bool supports_pretimeout = false;
```

Comment:
maybe make this a tristate int, i.e. initialize to -1 initialize, until we actually checked in which case its == 0 or > 1?